libopenage/util/file.cpp

```diff
@@ -122,13 +122,12 @@ std::ostream &operator<<(std::ostream &stream, const File &file) {
 	return stream;
 }
 
-static File get_temp_file() {
+File File::get_temp_file() {
 	fslike::Directory temp_dir = fslike::Directory::get_temp_directory();
 	std::string file_name = std::tmpnam(nullptr);
 	std::ostringstream dir_path;
 	temp_dir.repr(dir_path);
-	mode_t mode = 0777;
-	File file_wrapper = File(dir_path.str() + file_name, mode);
+	File file_wrapper = File(dir_path.str() + file_name, 0777);
```

Comment:
should not be 777! rather we need a flag if the tmp file should be executable, and have 644 and 755 accordingly.